### DISCUSSION TITLE
webRTC getUserMedia throws error

### DISCUSSION THREAD

**Discussion by syonip** [cefsharp/CefSharp#4221] [General]
Hi,
After getting intimidated by the text in "new issue" I decided to ask here.
I'm running the latest version of winform example, and loading this URL: https://webrtc.github.io/test-pages/src/audio-and-video/
Getting the error `User media request denied with error code 8`
<img width="862" alt="image" src="https://user-images.githubusercontent.com/5416329/185473710-3ba4b8b3-ad35-408b-bd97-fd589784dc6a.png">

Looking at the code, I see it's hooking a new `ExamplePermissionHandler` that looks like it's supposed to do something when a request is initiated. The breakpoints I've put in there aren't being hit.

I've added the code 
```
navigator.permissions.query({name: 'camera'})
 .then((permissionObj) => {
  console.log(permissionObj.state);
 })
 .catch((error) => {
  console.log('Got error :', error);
 })
```

And the permission state is being returned as "prompt".

Also tried adding the following lines which I saw mentioned in other issues:
```
settings.CefCommandLineArgs.Add("enable-media-stream", "1");
settings.CefCommandLineArgs.Add("enable-experimental-web-platform-features", "1");
settings.CefCommandLineArgs.Add("enable-usermedia-screen-capturing", "1");
```

Anyway, it seems this should work in the sample app out of the box no?

**Comment by amaitland** [comment]
If you test with https://github.com/cefsharp/CefSharp.MinimalExample how does it behave? `enable-media-stream` should be set by default.

If that doesn't work then testing with the [CEF Sample Application](https://github.com/cefsharp/CefSharp/blob/master/CONTRIBUTING.md#cefsharp-vs-chromium-embedded-frameworkcef) would be required. You'd also need to add `--enable-media-stream` to the command line.

`CEF` has only just added permissions handler support, so it's possible that not everything is working correctly yet. (`CefSharp` is just a `CEF` wrapper).

**Reply by amaitland** [reply]
> If that doesn't work then testing with the [CEF Sample Application](https://github.com/cefsharp/CefSharp/blob/master/CONTRIBUTING.md#cefsharp-vs-chromium-embedded-frameworkcef) would be required. You'd also need to add --enable-media-stream to the command line.

Quick test and the same alert is displayed, you'll need to move the conversation over to https://magpcss.org/ceforum/viewforum.php?f=6 as this is a `CEF` issue.

Errors from the log file:
```
[0821/132502.505:INFO:CONSOLE(0)] "Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.", source:  (0)
[0821/132504.937:INFO:render_frame_host_impl.cc(11654)] RenderFrameHostImpl::MaybeGenerateCrashReport url = https://webrtc.github.io/test-pages/src/audio-and-video/, status = 4, exit_code = 258
```

**Comment by syonip** [comment]
Actually I tried it now with the `--enable-media-stream` flag and getting the media stream works.
I thought setting `settings.CefCommandLineArgs.Add("enable-media-stream", "1");` would do the same, but you actually have to send a flag to the process.

There's probably still a bug with permissions because none of the permissions callbacks are invoked, but at least getting the stream works.
